### PR TITLE
SingleProcessorTopology implements Processor

### PR DESCRIPTION
**Pull request by guozhangwang**
@ymatsuda 

Let SingleProcessorTopology implements Processor so that uers do not need to pass-in the Processor class.

### CODE REVIEW

File: stream/src/main/java/org/apache/kafka/stream/topology/SingleProcessorTopology.java [hunk 1]
```diff
@@ -19,24 +19,17 @@
...
     @SuppressWarnings("unchecked")
     @Override
     public void topology() {
-        from(topics).process(newProcessor());
-    }
-
-    @SuppressWarnings("unchecked")
-    private Processor newProcessor() {
-        return (Processor) Utils.newInstance(processorClass);
+        from(topics).process((Processor)this);
```
Comment (ymatsuda) [inline stream/src/main/java/org/apache/kafka/stream/topology/SingleProcessorTopology.java]: @guozhangwang 
We need to create a new instance of Processor here, don't we?

Comment (guozhangwang) [inline stream/src/main/java/org/apache/kafka/stream/topology/SingleProcessorTopology.java]: @ymatsuda 
I think we can just use the SingleProcessorTopology instance here, why do we need to create another one?

Comment (ymatsuda) [inline stream/src/main/java/org/apache/kafka/stream/topology/SingleProcessorTopology.java]: topology() is called for each partition group to instantiate the chain of instances of KStream, Processor, StateStore, etc. In general, you shouldn't use the same Processor instance for different partition groups.

BTW, I don't know if we still want to do a special handing of a single processor use case. I am afraid that It confuses user.

Comment (guozhangwang) [inline stream/src/main/java/org/apache/kafka/stream/topology/SingleProcessorTopology.java]: You are right, I will go ahead and change accordingly.

**Pull request merged**